### ISSUE TITLE
Selecting on touch devices causes highlight flickering when move down rapidly 

### ISSUE THREAD

**Issue by oleksandr-danylchenko** [recogito/text-annotator-js#172]
## Issue
I noticed that when the `TextAnnotatorPopup` is rendered and I move select text down rapidly - the finger can go over the floating element and cause the highlight flickering:

https://github.com/user-attachments/assets/ebdb8383-c4f1-4f49-859e-1ffd7d1818b7

However, it doesn't happen when I remove the popup rendering: 

https://github.com/user-attachments/assets/9b208e65-39a2-49e6-a544-23018b4c66ed

---

In the real app on iOS it looks even more flickery when the selection goes over the headings: 

https://github.com/user-attachments/assets/3f60494a-1b35-4467-9339-5a73dd2757f3

**Comment by rsimon** [comment]
Yes, I saw this, too. But haven't investigated further yet. 

The selection clearly breaks as it goes from the selection start (on the text) to somewhere on the portal popup. I believe the popup is already marked as `not-annotatable`? Originally, I had the popup oriented into the upwards direction. In practice, that helps a bit since people will _usually_ want to select downward. (Although there's no guarantee, of course.) But because of the native context menu popup, downwards orientation really is the only option. Not sure what a good way around this would be...

**Comment by oleksandr-danylchenko** [comment]
>  But haven't investigated further yet

No problem, I'm on it now.

Here's the selection change event `range` breakdown:
![image](https://github.com/user-attachments/assets/c0123b73-451c-4f92-acfd-f032d4487b69)

When the jump occurs, the `SelectionHandler` thinks that you selected down to the end of the annotatable container. However, the range updates back to the "normal" value almost instantly.

**Comment by rsimon** [comment]
Thanks!! Hm, seems like the selection goes to the popup, which itself is not annotatable. Therefore, the clipping will select __up to__ the popup which, because the popup is portalled, is the end of the page?

**Comment by oleksandr-danylchenko** [comment]
> up to the popup which, because the popup is portalled, is the end of the page?

Right! It's exactly at the end of the page: 
![image](https://github.com/user-attachments/assets/6767dc04-f6ef-40d6-ad69-bc1b701627c7)

Therefore, the `SelectionHandler` just clamps the selection to the end of the annotatable container

**Comment by oleksandr-danylchenko** [comment]
It seems to be a common issue in the annotating solutions though 😅 
In Slab, they also render the floating popup below the selection and it's mounted _before_ the content in DOM. However, it's still possible to reach it when you select rapidly enough. That will make the selection jump up. Or when you scroll from the top - it will make the whole screen scroll up!

https://github.com/user-attachments/assets/2e732fbd-d9d1-43bb-a888-32dd9503972e

https://github.com/user-attachments/assets/97e8362f-f038-4833-94f4-b8bb40f21b65

**Comment by oleksandr-danylchenko** [comment]
I also tried putting the `user-select: none` across the `TextAnnotatorPopup`, however, it doesn't seem to be effective and the selection range can still go over it.

**Comment by rsimon** [comment]
Putting the portal at the top rather than the end sounds. But yikes for the scroll behavior... That seems actually worse UX than the jumping selection I believe.

> I also tried putting the user-select: none

Ouch. The selection still happens even if the popup has `user-select: none`, even with the various browser prefixes? I was thinking of this as a last resort, too. Keep it unselectable until the selection is idle for a while or so...

**Comment by oleksandr-danylchenko** [comment]
> The selection still happens even if the popup has `user-select: none`

That's correct. I tried to apply the `user-select: none` in various way, but none of them seem to prevent the selection over the popup ;(
I started with the following CSS, but it didn't yield any result 🤷🏻‍♂️ 
```css
.r6o-popup {
  ...
  user-select: none;
}
.r6o-popup * {
  user-select: none;
}
```

Then I tried the following, more aggressive `user-select: none` setting + the CSS above:
![image](https://github.com/user-attachments/assets/8e290ba2-7e08-43f0-b83b-98e393fe9196)

However, I still see the flickering when I select down rapidly.

**Comment by rsimon** [comment]
Agreed  - sounds sensible.

**Comment by oleksandr-danylchenko** [comment]
I tried adding the following simple hook for tracking the idling on a single annotation:
```ts
const useAnnotationTargetIdling = (
  annotationId: string | undefined,
  options: { timeout: number } = { timeout: 500 }
) => {
  const store = useAnnotationStore();

  const [isIdling, setIdling] = useState(true);

  useEffect(() => {
    if (!annotationId) return;

    let idlingTimeout: ReturnType<typeof setTimeout>;

    const scheduleSetIdling = () => {
      setIdling(false);

      clearTimeout(idlingTimeout);
      idlingTimeout = setTimeout(() => setIdling(true), options.timeout);
    };

    store.observe(
      scheduleSetIdling,
      {
        annotations: annotationId,
        ignore: Ignore.BODY_ONLY,
        origin: Origin.LOCAL
      }
    );
    return () => {
      clearTimeout(idlingTimeout);
      store.unobserve(scheduleSetIdling);
    };
  }, [annotationId]);

  return isIdling;
};
```

It works pretty nice for the cases when I do a continuous selection and then release the finger. It accounts for [the time needed](https://github.com/annotorious/annotorious/pull/467#issuecomment-2459832369) for the selector to adjust the range to the whitespace. 
But... it causes flicking in a case when I halted the selection, the idling got registered and then I released the finger. 

*loading the video in*

That's caused by another reported issue - https://github.com/recogito/text-annotator-js/issues/169. The store reports about the `target` update although there are no differences in the selected range!
